src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerTopPanel.xaml.cs

```diff
@@ -13,14 +18,145 @@ namespace NuGet.PackageManagement.UI
     /// </summary>
     public partial class PackageManagerTopPanel : UserControl
     {
-        private FilterLabel _selectedFilter;
+        private TabItem _selectedTabItem
+        {
+            get
+            {
+                return tabsPackageManagement.SelectedItem as TabItem;
+            }
+            set
+            {
+                tabsPackageManagement.SelectedItem = value;
+            }
+        }
+        public TabItem _tabConsolidate { get; private set; }
+        public Border _countConsolidateContainer { get; private set; }
+        public TextBlock _countConsolidate { get; private set; }
 
         public PackageManagerTopPanel()
         {
             InitializeComponent();
+        }
+
+        public void CreateAndAddConsolidateTab(bool isSolution)
+        {
+            IsSolution = isSolution;
+            if (IsSolution)
+            {
+                TabItem tabConsolidate = new TabItem();
```

Comment:
Here and below you can use `var` for `new` initializations.

Reply:
Done w/ 1 tradeoff...

Couldn't use `nameof `with textConsolidate's inline constructor variable initialization, so had to split it out, like `textConsolidate.Name = ...`.